sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/KeyVaultSettingsAsyncClient.java

```diff
@@ -0,0 +1,230 @@
+
+package com.azure.security.keyvault.administration;
+
+import com.azure.core.annotation.ReturnType;
+import com.azure.core.annotation.ServiceClient;
+import com.azure.core.annotation.ServiceMethod;
+import com.azure.core.http.rest.Response;
+import com.azure.core.http.rest.SimpleResponse;
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.security.keyvault.administration.implementation.KeyVaultAdministrationUtils;
+import com.azure.security.keyvault.administration.implementation.KeyVaultSettingsClientImpl;
+import com.azure.security.keyvault.administration.implementation.models.KeyVaultErrorException;
+import com.azure.security.keyvault.administration.implementation.models.Setting;
+import com.azure.security.keyvault.administration.models.KeyVaultSetting;
+import com.azure.security.keyvault.administration.models.KeyVaultSettingType;
+import reactor.core.publisher.Mono;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.azure.core.util.FluxUtil.monoError;
+
+/**
+ * The {@link KeyVaultSettingsAsyncClient} provides asynchronous methods to create, update, get and list
+ * {@link KeyVaultSetting settings} for the Azure Key Vault.
+ *
+ * <p>Instances of this client are obtained by calling the {@link KeyVaultSettingsClientBuilder#buildAsyncClient()}
+ * method on a {@link KeyVaultSettingsClientBuilder} object.</p>
+ *
+ * @see KeyVaultSettingsClientBuilder
+ */
+@ServiceClient(builder = KeyVaultSettingsClientBuilder.class, isAsync = true, serviceInterfaces =
+    KeyVaultSettingsClientImpl.KeyVaultSettingsClientService.class)
+public final class KeyVaultSettingsAsyncClient {
+    private final String vaultUrl;
+    private final KeyVaultSettingsClientImpl implClient;
+    private final ClientLogger logger = new ClientLogger(KeyVaultSettingsAsyncClient.class);
+
+    /**
+     * Creates a {@link KeyVaultSettingsAsyncClient} that uses a {@link KeyVaultSettingsClientImpl} to service requests.
+     *
+     * @param vaultUrl The URL of the key vault this client will act on.
+     * @param implClient The implementation client used to service requests.
+     */
+    KeyVaultSettingsAsyncClient(String vaultUrl, KeyVaultSettingsClientImpl implClient) {
+        this.vaultUrl = vaultUrl;
+        this.implClient = implClient;
+    }
+
+    /**
+     * Updates a given account setting with the provided value.
+     *
+     * @param name The name of the account setting to update.
+     * @param value The value to set.
+     *
+     * @return A {@link Mono} containing the updated {@link KeyVaultSetting account setting}.
+     *
+     * @throws IllegalArgumentException thrown if {@code name} or {@code value} is {@code null} or empty.
+     * @throws KeyVaultErrorException thrown if the request is rejected by the server.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<KeyVaultSetting> updateSetting(String name, String value) {
```

Comment:
I don't think we should expose `String value` since right now these are only booleans supported. In .NET, I made this an overload for which we can add more types later. We need to decide how we want to expose these. For example, maybe we should make `KeyVaultSetting` read/write with helpers to set `Value` without exposing `Value` directly and make `Type` read-only (name should be read-only too). Not all languages support overloads so using an overload for `UpdateSetting` may not be ideal if we want some consistency.

Reply:
Looking at AppConfig, seems they don't do anything special but I wonder if we should at least provide helpers in some fashion: https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationSetting.cs

For example, if a customer were to use `bool.ToString()` in .NET, it actually returns (sadly) "True" and "False". Would the service even accept those values or would it require "true" and "false" if case-sensitive? What about serializing integers? Someone might use just `int.ToString()` which will serialize differently for different locales unless you pass an invariant culture, which they also might not realize.

Reply:
I believe, though, this was more because AppConfig could store just about anything that could be serialized as a string, so content-type is rather open-ended. In our case, it's more of a token type, like you'd see in a JSON parser.

Reply:
> I don't think we should expose `String value` since right now these are only booleans supported. In .NET, I made this an overload for which we can add more types later. We need to decide how we want to expose these. For example, maybe we should make `KeyVaultSetting` read/write with helpers to set `Value` without exposing `Value` directly and make `Type` read-only (name should be read-only too). Not all languages support overloads so using an overload for `UpdateSetting` may not be ideal if we want some consistency.

The `KeyVaultSetting` type model for Java would be a return type and not something users can pass to our APIs. I think we're fine letting the `name` be a part of the constructor as a `String` as users shouldn't be instantiating the class anyways. As for the `updateSetting()` method, we can add an overload for `boolean` but I think we should definitely accept `value` as a `String` at the very least.